tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,699 @@
+	return services, err
+}
+
+// Find returns true if val exists in the slice array, false otherwise
+func Find(slice []string, val string) bool {
+	for _, item := range slice {
+		if item == val {
+			return true
+		}
+	}
+	return false
+}
+
+func deployCluster(customValues string, helmValues map[string]string) {
+	clusterChartPath, err := filepath.Abs("../../charts/k8ssandra")
+	if err != nil {
+		Fail("Couldn't find the absolute path for K8ssandra charts")
+	}
+
+	customChartPath, err := filepath.Abs("charts/" + customValues)
+	if err != nil {
+		Fail(fmt.Sprintf("Couldn't find the absolute path for custom values: %s", customValues))
+	}
+
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	helmOptions := &helm.Options{}
+
+	helmOptions = &helm.Options{
+		// Enable traefik to allow redirections for testing
+		SetValues:      helmValues,
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+		ValuesFiles:    []string{customChartPath},
+	}
+
+	releaseName := "k8ssandra"
+	defer timeTrack(time.Now(), fmt.Sprintf("Installing and starting k8ssandra"))
+	helm.Install(GinkgoT(), helmOptions, clusterChartPath, releaseName)
+
+	// Wait for cass-operator pod to be ready
+	attempts := 0
+	maxAttempts := 10
+	for {
+		attempts++
+		clientset, _ := k8s.GetKubernetesClientFromOptionsE(GinkgoT(), kubectlOptions)
+		pods, _ := clientset.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app.kubernetes.io/name=cass-operator"})
+		if len(pods.Items) == 1 {
+			k8s.RunKubectl(GinkgoT(), kubectlOptions, "wait", "--for=condition=Ready", "pod", "-l", "app.kubernetes.io/name=cass-operator", "--timeout=1800s")
+			break
+		} else if attempts > maxAttempts {
+			Fail("Couldn't find cass-operator pod")
+		}
+		time.Sleep(20 * time.Second)
+	}
+
+	// Wait for CassandraDatacenter to be ready..
+	k8s.RunKubectl(GinkgoT(), kubectlOptions, "wait", "--for=condition=Ready", "cassandradatacenter/dc1", "--timeout=1800s")
+}
+
+func GetUsernamePassword(secretName, ns string) credentials {
+	kubectlOptions := k8s.NewKubectlOptions("", "", ns)
+	secret := k8s.GetSecret(GinkgoT(), kubectlOptions, secretName)
+	username := secret.Data["username"]
+	password := secret.Data["password"]
+	creds := credentials{string(username), string(password)}
+	return creds
+}
+
+func runCassandraQueryAndGetOutput(query string) string {
+	cqlCredentials := GetUsernamePassword("k8ssandra-superuser", namespace)
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	// Get reaper service
+	output, _ := k8s.RunKubectlAndGetOutputE(GinkgoT(), kubectlOptions, "exec", "-it", "k8ssandra-dc1-default-sts-0", "--", "/opt/cassandra/bin/cqlsh", "--username", cqlCredentials.username, "--password", cqlCredentials.password, "-e", query)
+	return output
+}
+
+func WaitForPodWithLabelToBeReady(label string, waitTime time.Duration, maxAttempts int) {
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	attempts := 0
```

Comment:
This whole function should be the last command and only last command.

Reply:
Yup, moving all these retry based checks to use `Eventually` now.